halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/deck/DeckProfileFactory.java

```diff
@@ -232,8 +234,35 @@ protected void setProfile(
       bindings.put("canary.showAllCanaryConfigs", canary.isShowAllConfigsEnabled());
     }
 
+    // Configure Plugins
```

Comment:
This piece is how we send plugin configuration (name, location, and soon type) to deck, so it can create tags for downloading these resources. 

This PR uses init containers to download those resources, however if we don't utilize init containers, then we can just use the script tags to download resources where they're hosted (eg vs downloading them so Deck can host the resources).